app.territoiresentransitions.react/src/app/pages/collectivite/Indicateurs/useIndicateurDefinitions.ts

```diff
@@ -0,0 +1,172 @@
+import {useQuery} from 'react-query';
+import {DISABLE_AUTO_REFETCH, supabaseClient} from 'core-logic/api/supabase';
+import {useCollectiviteId} from 'core-logic/hooks/params';
+import {TIndicateurDefinition, TIndicateurReferentielDefinition} from './types';
+import {ReferentielOfIndicateur} from 'types/litterals';
+
+/** Charge et cache les définitions de tous les indicateurs */
+export const useIndicateurDefinitions = () => {
+  const {data} = useQuery(
+    ['indicateur_definition'],
+    async () => {
+      const {data, error} = await supabaseClient
+        .from('indicateur_definition')
+        .select('*, actions: indicateur_action (action_id)')
+        .order('nom', {ascending: true});
+
+      if (error) {
+        throw new Error(error.message);
+      }
+
+      return (
+        data
+          // tri par nom (pour que les diacritiques soient pris en compte)
+          ?.sort((a, b) => a.nom.localeCompare(b.nom))
+          // extrait les ids des actions liées aux indicateurs
+          .map(definition => {
+            const {actions} = definition;
+            if (Array.isArray(actions) && actions?.length) {
+              return {...definition, actions: actions.map(a => a.action_id)};
+            }
+            return {...definition, actions: null};
+          }) as TIndicateurReferentielDefinition[]
+      );
+    },
+    DISABLE_AUTO_REFETCH
+  );
+  return data || [];
+};
+
+/** Fourni les définitions des indicateurs "enfants" à partir de l'id du parent */
+export const useIndicateursEnfants = (parentId: string) =>
+  useIndicateurDefinitions()?.filter(({parent}) => parent === parentId);
+
+/** Fourni les définitions des indicateurs "parents" */
+export const useIndicateursParents = () =>
+  useIndicateurDefinitions()?.filter(({parent}) => parent === null);
+
+/** Fourni les définitions des indicateurs par groupe d'indicateurs prédéfinis */
+export const useIndicateursParentsGroup = (group: ReferentielOfIndicateur) => {
+  // on affiche que les indicateurs parents sauf dans le cas CRTE
+  const tous = useIndicateurDefinitions();
+  const parents = useIndicateursParents();
+  return (group === 'crte' ? tous : parents)?.filter(({programmes}) =>
```

Comment:
En local, collectivité 1, j'ai une erreur sur cette ligne en chargeant la page d'accueil, due à l'appel de `useIndicateursCount`, et donc de `useIndicateursParentsGroup` depuis le composant `<IndicateursCard>`.

`Uncaught TypeError: Cannot read properties of undefined (reading 'includes')`

J'ai de nouveau l'erreur en allant dans indicateurs cae, depuis le composant `<IndicateursPredefinisList>`.

Reply:
Tu as testé avec la bonne version du back ? C'est un changement récent (`indicateur_groupe` a été déprécié au profit de `programmes` avec lequel il faisait doublon)

Reply:
En effet je n'ai plus d'erreur, my bad 😬